src/backend/distributed/planner/distributed_planner.c

```diff
@@ -99,6 +100,19 @@ static void ResetPlannerRestrictionContext(
 static bool HasUnresolvedExternParamsWalker(Node *expression, ParamListInfo boundParams);
 
 
+typedef struct inline_cte_walker_context
```

Comment:
`inline_cte_walker_context `, `inline_cte_walker `, `contain_dml ` and `contain_dml_walker ` copy paste from postgres.


`inline_cte ` and `InlineCTEs ` are heavily based on Postgres code